### ISSUE TITLE
Crash in forbid-prop-types

### ISSUE THREAD

**Issue by epmatsw** [yannickcr/eslint-plugin-react#230]
I'm getting a crash on line 65 of forbid-prop-types.js. `Cannot read property 'name' of undefined`. My guess is that the line above (`declaration.value = declaration.value.object;`) is storing an object without a property value, which then crashes. It could probably be fixed by just doing an existence check, but I'm not entirely sure that there isn't an upstream issue that's the root cause.

**Comment by epmatsw** [comment]
I'm also unfortunately not sure what code exactly is triggering this. It's coming from somewhere in our `gulp-eslint` process, which makes it a bit hard to track down :/

**Comment by yannickcr** [comment]
Thanks for the report, I'll try to fix this ASAP.

It would be a great help for us if you can track down the code responsible for the crash, maybe by running ESLint on one directory at a time until you find the file that trigger this error ?

**Comment by epmatsw** [comment]
I think I've figured it out. We're destructuring React to get {PropTypes}, so the propTypes that triggers the crash looks like:
```
propTypes: {
        items: PropTypes.arrayOf(PropTypes.object).isRequired,
        renderCard: PropTypes.func.isRequired,
        actions: PropTypes.object,
        multiActions: PropTypes.object
      }
```

I would guess that the plugin expects only React.PropTypes, which causes the error.

**Comment by epmatsw** [comment]
Then again, maybe not. Changing to React.PropTypes did not seem to fix it.

**Comment by chollier** [comment]
I'm getting the same issue, this code makes it crash :

```javascript
  static propTypes = {
    retailer: PropTypes.instanceOf(Map).isRequired,
    requestRetailer: PropTypes.func.isRequired
  }
```
this code doesn't :
```
  static propTypes = {
    // retailer: PropTypes.instanceOf(Map).isRequired,
    requestRetailer: PropTypes.func.isRequired
  }
```

**Comment by epmatsw** [comment]
Not entirely fixed it seems.

```
React.createClass({
  propTypes: {
    options: optionsType
  }
});
```

**Comment by naddeoa** [comment]
Found another example to contribute. This doesn't work:

```javascript
var optionPropType = {
    value: React.PropTypes.string.isRequired,
    label: React.PropTypes.string.isRequired
};

var DropdownView = React.createClass({
    propTypes: {
        widthNamed: Dropdown.propTypes.width,
        optionPrompt: React.PropTypes.string,
        selected: optionPropType,
        options: React.PropTypes.arrayOf(React.PropTypes.shape(optionPropType))
    }
});
```

Then, I noticed that I was actually passing in a plain object to `propTypes.selected`. I made `optionPropType ` a `React.propType.shape` and that fixed it:
```javascript
var optionPropType = React.PropTypes.shape({
    value: React.PropTypes.string.isRequired,
    label: React.PropTypes.string.isRequired
});

var DropdownView = React.createClass({
    propTypes: {
        widthNamed: Dropdown.propTypes.width,
        optionPrompt: React.PropTypes.string,
        selected: optionPropType,
        options: React.PropTypes.arrayOf(optionPropType)
    }
});
```
